text/1003-cli-naming.rst

```diff
@@ -19,110 +19,238 @@ specified.
+or of the currently running server, or of how many active connections the
+server has at the moment. Renaming it to ``edgedb show-migration-status``
+would make it longer to type and overall very verbose. With
+``<group> <command>`` scheme the solution is obvious:
+``edgedb migration status``.
 
-4. The ``<action>-<category>`` scheme does not allow category-specific
-   help sections.
+**Decision**: ``edgedb show-status`` should be renamed to
+``edgedb migration-status`` which is consistent with
+``edgedb migration-log``.
 
-   With ``<group> <command>`` scheme, top-level help sections are natural:
-   ``edgedb migration -h`` command can have a description of the recommended
-   migrations workflow, while ``edgedb server -h`` would explain to the user
-   how to work with EdgeDB servers.
+In general, vague nouns and verbs should be avoided unless
+the feature in question is dead obvious (like ``edgedb dump``,
+``edgedb query``, or ``edgedb server init``).
 
-5. ``edgedb configure`` currently implements the ``<group> <command>`` scheme;
-   it should have been ``edgedb insert-config``, ``edgedb set-config``,
-   ``edgedb reset-config``.
+This will also future-proof the CLI for new kinds of functionality that
+would use the same vague terms (e.g. ``edgedb time-machine-log``).
 
-6. The ``edgedb server`` CLI design outlined in RFC 1001 directly conflicts
-   with the ``<action>-<category>`` scheme.
+Category-specific help sections
+-------------------------------
 
-The last observation is what prompted the creation of this RFC.  We either
-need to update RFC 1001 to use the ``<action>-<category>`` scheme or
-adopt the ``<group> <command>`` for all other commands.  Combining two
-drastically different CLI designs in one tool is not an option.
+**Concern**: The ``<action>-<category>`` scheme does not allow
+category-specific help sections. With ``<group> <command>`` scheme, top-level
+help sections are natural: ``edgedb migration -h`` command can have a
+description of the recommended migrations workflow, while
+``edgedb server -h`` would explain to the user how to work with EdgeDB
+servers.
 
+**Decision**: the CLI is an unlikely place for a user to learn about new
+complex functionality like the migration subsystem. Better suited avenues
+for this are the online documentation and/or ``man`` pages, or better yet,
+``tldr``.
 
-Overview
-========
+To address discoverability of related sub-commands, top-level listing of
+sub-commands should not be alphabetical but rather grouped by functionality.
 
-The RFC proposes to adopt the ``<group> <command>`` naming scheme for
-the ``edgedb`` CLI.  Here's an outline of current and future ``edgedb``
-subcommands:
+``configure`` uses ``<group> <command>`` whereas ``list-`` is flat
+------------------------------------------------------------------
+
+**Concern**: ``configure`` is inconsistent with ``list-``.
+
+**Decision**: It should be replaced by ``edgedb insert-config``,
```

Comment:
ACTIONABLE ITEM 4: rename `configure`.

Reply:
I'm OK with this, although I'm also OK to keep it as is. @elprans ?